go/libraries/doltcore/dbfactory/aws.go

```diff
@@ -188,6 +193,11 @@ func awsConfigFromParams(params map[string]interface{}) (session.Options, error)
 		opts.Profile = val.(string)
 	}
 
+	filePath, ok := params[AWSCredsFileParam]
+	if ok && len(filePath.(string)) != 0 && awsCredsSource == RoleCS {
+		awsCredsSource = FileCS
+	}
+
```

Comment:
If a credential file is provided then change the source to FileCS unless it has been explicitly changed from the default of RoleCS